### PR TITLE
Enable FindBugs in the plugin

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Enables FindBugs checks in https://jenkins.ci.cloudbees.com/job/Plugins/job/unique-id-plugin/

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@reviewbybees

File: src/main/java/org/jenkinsci/plugins/uniqueid/impl/IdStoreMigratorV1ToV2.java [hunk 1]
```diff
@@ -65,14 +65,7 @@ static void performMigration(@Nonnull Jenkins jenkins) {
         List<Item> allItems = jenkins.getAllItems();
 
         for (Item item : allItems) {
-            // can only be Folder or Job here (not a run) - and these both implement PersistenceRoot
-            if (item instanceof PersistenceRoot) {
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/uniqueid/impl/IdStoreMigratorV1ToV2.java]: Any Item implements PersistenceRoot

File: pom.xml [hunk 2]
```diff
@@ -56,6 +56,12 @@
       <version>1.8.3</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.google.code.findbugs</groupId>
+      <artifactId>annotations</artifactId>
+      <version>3.0.0</version>
+      <type>jar</type>
```
Comment (jtnord) [inline pom.xml]: type is default - so don;t specify.
Needs to be scope provided so it isn't included with the bundle (and possibly optional=true as well)

File: pom.xml [hunk 3]
```diff
@@ -87,6 +93,16 @@
           <compatibleSinceVersion>2.0</compatibleSinceVersion>
         </configuration>
       </plugin>
+      <plugin>
```
Comment (jtnord) [inline pom.xml]: as this is one of our plugins - I would like to go further and make sure findbugs is always run (locally) and not just on the CI system

File: pom.xml (hunk 2 above)
Comment (oleg-nenashev) [inline pom.xml]: Sorry, copy-pased without a check

**Comment by oleg-nenashev** [comment]
@jtnord Added FindBugs to verify phase

File: pom.xml [hunk 4]
```diff
@@ -87,6 +93,24 @@
...
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>findbugs-maven-plugin</artifactId>
+        <version>3.0.1</version>
+        <configuration>
+          <xmlOutput>true</xmlOutput>
+          <findbugsXmlWithMessages>true</findbugsXmlWithMessages>
+          <failOnError>false</failOnError>
+        </configuration>
+        <executions>
+          <execution>
```
Comment (jtnord) [inline pom.xml]: I would prefer in future that executions have an id

**Comment by jtnord** [comment]
:bee:

File: pom.xml (hunk 4 above)
Comment (oleg-nenashev) [inline pom.xml]: Done

**Comment by jtnord** [comment]
:bee: again

**Pull request merged**